app/models/user.rb

```diff
@@ -22,21 +22,32 @@
 #
 class User < ApplicationRecord
 
+  # TODO: Move this to Global Application Config class
+  THEMES ||= %w[system dark light].freeze
+
+  after_initialize :set_default_preferences, if: :new_record?
   before_save :sanitize_names
 
+  serialize :preferences, coder: HashSerializer
+  store_accessor :preferences, :theme, prefix: :preference
+  validates :preferences, preferences: true
+
   validates :email, presence: true, uniqueness: true
```

Comment:
We also need to validate email with email regex. Also use case_insensitive uniqueness.

Reply:
can be done in separate PR

Reply:
Yes let me open a follow up PR for this, thanks

Reply:
https://github.com/abhiyaantrix/event-radar/pull/19